server/container/guice/mailet/src/main/java/org/apache/james/modules/server/CamelMailetContainerModule.java

```diff
@@ -122,6 +129,17 @@ public MailetModuleConfigurationPerformer(ConfigurationProvider configurationPro
             this.mailQueueFactory = mailQueueFactory;
             this.transportProcessorCheckSet = transportProcessorCheckSet;
             this.defaultProcessorsConfigurationSupplier = defaultProcessorsConfigurationSupplier;
+            this.executorService = new SandglassExecutorService(getInitializationTimeout());
+        }
+
+        @VisibleForTesting Duration getInitializationTimeout() {
+            try {
+                return Duration.ofMillis(configurationProvider.getConfiguration("mailetcontainer")
+                        .getLong("initTimeoutInMillis", ONE_MINUTE_IN_MILLIS));
+            } catch (ConfigurationException e) {
+                LOGGER.warn("Could not locate configuration for initTimeoutInMillis. Default to one minute.");
```

Comment:
I don't understand why we need that. What can make init long ? Why it's not needed in other places ? Do we really want to make James fail on slow systems ?